### PR TITLE
Refactor TableMetadataManager

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/common/TableMetadataManager.java [hunk 1]
```diff
@@ -20,6 +21,14 @@ public class TableMetadataManager {
   private final LoadingCache<TableKey, Optional<TableMetadata>> tableMetadataCache;
 
   public TableMetadataManager(Admin admin, long cacheExpirationTimeSecs) {
+    this(
+        (key) -> Optional.ofNullable(admin.getTableMetadata(key.namespace, key.table)),
+        cacheExpirationTimeSecs);
+  }
+
+  public TableMetadataManager(
+      ThrowableFunction<Optional<TableMetadata>, TableKey, Exception> getTableMetadataFunc,
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/common/TableMetadataManager.java:30]: After this change, we can also specify a function to retrieve a table metadata in `TableMetadataManager`.

**Review by Torch3333 — DISMISSED** [review]
LGTM, thank you. 
I am curious, what is the end goal of this refactoring?

**Comment by brfrn169** [comment]
> I am curious, what is the end goal of this refactoring?

Sorry for the lack explanation. Actually, I wanted to use the new constructor in another part as follows:

```java
TableMetadataManager tableMetadataManager = new TableMetadataManager(
    key -> {
      // Retrieve table metadata
      ...
    },
    cacheExpirationTimeSecs);
```

We can pass only the `Admin` interface in the existing constructor, but it didn't suit my case.

**Comment by Torch3333** [comment]
@brfrn169 I see, thank you for the details.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**